### PR TITLE
Ce/hide reports

### PR DESCRIPTION
**Pull request by calellowitz**
@benrudolph 
This is to hide unused reports for icds. hold off merge for qa

### CODE REVIEW

File: corehq/apps/reports/management/commands/edit_report_whitelist.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        parser.add_argument(
+            '--reset',
+            action='store_true',
+            dest='reset',
+            default=False,
+            help='Clear whitelist before adding new reports.'
+        )
+
+
+    def handle(self, domain, reports, **options):
```
Comment (dimagimon) [inline corehq/apps/reports/management/commands/edit_report_whitelist.py]: E303 too many blank lines (2)

File: corehq/apps/reports/management/commands/edit_report_whitelist.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        reset = options.pop('reset')
+        domain = Domain.get_by_name(domain)
+
+        if reset:
+            domain.report_whitelist = []
+
+        for report in reports:
+            domain.report_whitelist.append(report)
+        domain.save()
```
Comment (dimagimon) [inline corehq/apps/reports/management/commands/edit_report_whitelist.py]: W292 no newline at end of file

File: corehq/reports.py [hunk 3]
```diff
@@ -136,13 +148,19 @@ def REPORTS(project):
...
     messaging_reports += getattr(Domain.get_module_by_name(project.name), 'MESSAGING_REPORTS', ())
+    messaging_reports = _filter_reports(report_set, messaging_reports)
     messaging = (ugettext_lazy("Messaging"), messaging_reports)
     reports.append(messaging)
 
     reports.extend(_get_dynamic_reports(project))
 
     return reports
 
+def _filter_reports(report_set, reports):
```
Comment (dimagimon) [inline corehq/reports.py]: E302 expected 2 blank lines, found 1

File: corehq/apps/reports/management/commands/edit_report_whitelist.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
+from django.core.management import BaseCommand
+
+from corehq.apps.domain.models import Domain
+
+
+class Command(BaseCommand):
+    help = 'Reindex a pillowtop index'
```
Comment (benrudolph) [inline corehq/apps/reports/management/commands/edit_report_whitelist.py]: nit: wrong help message

File: corehq/apps/reports/management/commands/edit_report_whitelist.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+    def handle(self, domain, reports, **options):
+        reset = options.pop('reset')
+        domain = Domain.get_by_name(domain)
+
+        if reset:
+            domain.report_whitelist = []
+
+        for report in reports:
+            domain.report_whitelist.append(report)
```
Comment (benrudolph) [inline corehq/apps/reports/management/commands/edit_report_whitelist.py]: report in this instance is the report slug?

**Review by benrudolph — APPROVED** [review]
only a few nitpicks, looks good

File: corehq/apps/reports/management/commands/edit_report_whitelist.py (hunk 5 above)
Comment (calellowitz) [inline corehq/apps/reports/management/commands/edit_report_whitelist.py]: yep. sorry, its a report slug. you just pass the needed slugs as arguments to the management command

**Pull request merged**